packages/dds/tree/src/test/shared-tree-core/sharedTreeCore.rollback.spec.ts

```diff
@@ -0,0 +1,198 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "node:assert";
+
+import { createIdCompressor } from "@fluidframework/id-compressor/internal";
+
+import {
+	MockContainerRuntimeFactory,
+	MockFluidDataStoreRuntime,
+	MockStorage,
+} from "@fluidframework/test-runtime-utils/internal";
+
+import { SharedTreeTestFactory } from "../utils.js";
+
+import { SchemaFactory, TreeViewConfiguration } from "../../simple-tree/index.js";
+import { FlushMode } from "@fluidframework/runtime-definitions/internal";
+
+const sf = new SchemaFactory("Test");
+class TestNode extends sf.objectRecursive("test node", {
+	child: sf.optionalRecursive([sf.number]),
+}) {}
+
+function setupTree() {
+	const containerRuntimeFactory = new MockContainerRuntimeFactory({
+		flushMode: FlushMode.TurnBased,
+	});
+	const dataStoreRuntime1 = new MockFluidDataStoreRuntime({
+		idCompressor: createIdCompressor(),
+	});
+
+	const factory = new SharedTreeTestFactory(() => {});
+
+	const containerRuntime = containerRuntimeFactory.createContainerRuntime(dataStoreRuntime1);
+	const tree1 = factory.create(dataStoreRuntime1, "A");
+	tree1.connect({
+		deltaConnection: dataStoreRuntime1.createDeltaConnection(),
+		objectStorage: new MockStorage(),
+	});
+
+	const view = tree1.viewWith(new TreeViewConfiguration({ schema: TestNode }));
+	view.initialize(new TestNode({}));
+	containerRuntime.flush();
+	return { view, containerRuntime, containerRuntimeFactory };
+}
+
+describe("SharedTreeCore rollback", () => {
```

Comment:
these test a pretty rudimentary. as i call out in the description, i plan to bring on more testing, but is there anything that should definitely be in the unit tests? ideally if someone breaks with flow dramatically these test catch it and are easier to debug than our model-based tests.

Reply:
Is there a way for a peer with unflushed changes to first learn of (concurrent) peer changes then have its unflushed changes rolled back? Is so, that seems good to test.